src/syrupy/serializers/amber.py

```diff
@@ -129,31 +147,52 @@ def serialize_dict(cls, data: "SerializableData", indent: int = 0) -> str:
         )
 
     @classmethod
-    def serialize_iterable(cls, data: "SerializableData", indent: int = 0) -> str:
+    def serialize_iterable(
+        cls, data: "SerializableData", *, indent: int = 0, visited: List[Any] = []
+    ) -> str:
         open_paren, close_paren = next(
             paren[1]
             for paren in {list: "[]", tuple: "()", GeneratorType: "()"}.items()
             if isinstance(data, paren[0])
         )
         return (
             cls.with_indent(f"{cls.object_type(data)} {open_paren}\n", indent)
-            + "".join([f"{cls.serialize(d, indent + 1)},\n" for d in data])
+            + "".join(
+                [
+                    f"{cls.serialize(d, indent=indent + 1, visited=visited)},\n"
+                    for d in data
+                ]
+            )
             + cls.with_indent(close_paren, indent)
         )
 
     @classmethod
-    def serialize(cls, data: "SerializableData", indent: int = 0) -> str:
+    def serialize_unknown(
+        cls, data: Any, *, indent: int = 0, visited: List[Any] = []
+    ) -> str:
+        return cls.with_indent(repr(data), indent)
+
+    @classmethod
+    def serialize(
+        cls, data: "SerializableData", *, indent: int = 0, visited: List[Any] = []
```

Comment:
should use a Set instead of List

Reply:
I think that requires each object to be hashable, which might not be feasible

Reply:
You could store the id in the set

Reply:
The id from hash?

Reply:
Id function. Basically store the memory addresses in a set since they're integers and thus always hashable. We're keeping everything in memory anyway

Reply:
Good point

Reply:
Updated